Exposing trace record meta info in the task context (#5402)
nextflow-io/nextflow: docs/process.md

The directive {ref}`process-maxretries` set the maximum number of time the same task can be re-executed.

### Dynamic task resources with previous execution trace
:::{versionadded} 24.09.1-edge
:::
Similarly to the previous scenario, task resources can be also updated according to the metrics included in the {ref}`trace record <trace-report>` of the previous task attempt. The metrics can be accessed through the `trace` variable. For example:

```groovy
process foo {

>     memory { (trace != null) ? trace.memory * 2 : (1.GB) }

Reviewer: ```suggestion
    memory { task.attempt > 1 ? trace.memory * 2 : (1.GB) }
```

The directive {ref}`process-maxretries` set the maximum number of time the same task can be re-executed.

### Dynamic task resources with previous execution trace

> :::{versionadded} 24.09.2-edge

Reviewer 2: ```suggestion
:::{versionadded} 24.10.0
```

:::

> Similarly to the previous scenario, task resources can be also updated according to the metrics included in the {ref}`trace record <trace-report>` of the previous task attempt. The metrics can be accessed through the `task.previousTrace` variable. For example:

Reviewer 3: ```suggestion
Task resource requests can be updated relative to the {ref}`trace record <trace-report>` metrics of the previous task attempt. The metrics can be accessed through the `task.previousTrace` variable. For example:
```

```groovy
process foo {
    memory { task.attempt > 1 ? task.previousTrace.memory * 2 : (1.GB) }
    errorStrategy { task.exitStatus in 137..140 ? 'retry' : 'terminate' }
    maxRetries 3

    script:
    <your job here>
}
```

> In the above example the {ref}`process-memory` is set according to previous trace record metrics. In the first attempt, when no trace metrics are available, it is set to one GB, while in the other attempts it is doubling the previously allocated memory. The available metrics in the trace record can be found in the {ref}`Trace Report section <trace-report>`.

Reviewer 3: ```suggestion
In the above example, the {ref}`process-memory` is set according to previous trace record metrics. In the first attempt, when no trace metrics are available, it is set to one GB. In the subsequent attempts, it doubles the previously allocated memory. See {ref}`trace-report` for more information about trace records.
```
